pkg/curatedpackages/bundleclient.go

```diff
@@ -0,0 +1,172 @@
+package curatedpackages
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	packagesv1 "github.com/aws/eks-anywhere-packages/api/v1alpha1"
+	"github.com/aws/eks-anywhere-packages/pkg/artifacts"
+	"github.com/aws/eks-anywhere-packages/pkg/bundle"
+	"github.com/aws/eks-anywhere/pkg/constants"
+)
+
+// BundleClient abstracts the source of package bundle information.
+//
+// Whether the source is a registry, a cluster, the BundleClient should
+// abstract the details of retrieving a particular curated packages bundle.
+type BundleClient interface {
+	ActiveOrLatest(ctx context.Context) (*packagesv1.PackageBundle, error)
```

Comment:
nit: 
```suggestion
	GetActiveOrLatestBundle(ctx context.Context) (*packagesv1.PackageBundle, error)
```

Reply:
Bundle is repetitive. The package is bundleclient, so bundleclient.ActiveOrLatestBundle is a form of stuttering. I'd prefer to avoid it. Same for Get, it's implied. If you need an authority: https://go.dev/doc/effective_go#names

Reply:
If this truly is a nit, I think I'll stand by my choices. I prefer the lack of repetition and generally find "Get*" to be implied. https://go.dev/doc/effective_go#Getters

https://go.dev/doc/effective_go#package-names refers more to package names, but I think the same case can be made, ie that bundleclient.ActiveOrLatestBundle is a form of repetition that doesn't add any value over bundleclient.ActiveOrLatest. (Plus, if you're in doubt about what type it returns, since Go is a typed language, your editor or the compiler can easily tell you that information with greater certainty, and it's never out of date.)

Reply:
It's a nit, so you can stand by your choice :)

That said, my suggestion to add `Get` is not because this is a getter (as in an accessor to a field), I would agree with you, those don't need the word `Get`. But this method is actually going and getting something for somewhere else. And I believe representing that in the method name gives context to the caller about the possible cost of calling this multiple times (vs calling an accessor multiple times, which is mostly free).

About the `Bundles` part, I see what you are saying. However, when I read `bundlesclient` I'm not completely sure if that client can only return a bundle or it might be able to return other things. That in addition with the absence of the word `Get` (or any other synonym) create ambiguity. `bundleclient.ActiveOrLatest` could very much return the active or latest client.

I'm all in favor of not using redundant names and try to keep them short, that makes the code easier to read. But in this case, IMO, we are taking it a bit to far and actually making the code more difficult to read.